dbt/adapters/default.py

```diff
@@ -0,0 +1,527 @@
+import copy
+import multiprocessing
+import re
+import time
+import yaml
+
+from contextlib import contextmanager
+
+import dbt.exceptions
+import dbt.flags
+
+from dbt.contracts.connection import validate_connection
+from dbt.logger import GLOBAL_LOGGER as logger
+from dbt.schema import Column
+
+
+lock = multiprocessing.Lock()
```

Comment:
hah i wouldn't worry about multiple connection types for now. That sounds right to me, just wanted to confirm because we had some problems with single-imports back in the day